master/internal/rm/kubernetes/pods.go

```diff
@@ -232,8 +242,49 @@ func (p *pods) Receive(ctx *actor.Context) error {
 	return nil
 }
 
+func readClientConfig(credsDir string) (*rest.Config, error) {
+	if credsDir == "" {
+		// The default in-cluster case.  Internally, k8s.io/client-go/rest is going to look for
+		// environment variables:
+		//   - KUBERNETES_SERVICE_HOST
+		//   - KUBERNETES_SERVICE_PORT
+		// and it expects to find files:
+		//   - /var/run/secrets/kubernetes.io/serviceaccount/token
+		//   - /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+		return rest.InClusterConfig()
+	}
+
+	// A special case for rapid determined+k8s development: build a rest.Config from a specially
+	// packed directory with the same information.  Our tools/scripts/fetch-k8s-creds.sh script can
+	// create such a directory, with server, token, and ca.crt files.
+
+	//nolint:gosec // XXX I don't really understand why this is risky.
+	server, err := ioutil.ReadFile(filepath.Join(credsDir, "server"))
```

Comment:
I'd prefer to just nest a `rest.Config` in the k8s config as `_apiserver_config` and then keep this dev-specific junk in a shell script. but this is also fine, totally non-blocking, just stating my preference.

Reply:
I can see the appeal of that, from the master code perspective, but I think it makes it harder for a human to actually interact with this dev feature.